### DISCUSSION TITLE
Should we add wrap-and-create functions in "Batch"?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/v2-periphery#225] [Ideas]
As we decided that we will no longer use proxy targets in `2.1`, we won't have the `wrapAndCreate` helper functions from targets. Should we add support for them in the `SablierV2Batch` contract? @PaulRBerg

@razgraf @gavriliumircea, are they used in the app?

**Comment by razgraf** [comment]
We're not using them in the app, *yet*. So there's no rush in adding them as far as the current client functionality goes.

On the other hand, regarding deployment cost, it may make sense to add them now rather than later.

**Reply by andreivladbrg** [reply]
>On the other hand, regarding deployment cost, it may make sense to add them now rather than later.

yep, agree

**Reply by PaulRBerg** [reply]
It would be nice to add them but we can't do it, unfortunately, because the audit for `Batch` would be invalidated.

**Comment by PaulRBerg** [comment]
Great observation @andreivladbrg - I'm in favor of adding them in V2.2 (see my comment above).

The other wrinkle is naming. Does it make sense to add these functions in a contract called `Batch`?

We might want to rename it to something else.

**Reply by andreivladbrg** [reply]
> The other wrinkle is naming. Does it make sense to add these functions in a contract called Batch?

yes, agree, the naming is problematic, but also, deploying a new set of contracts (on all chain) just for these wrapper functions would be wasteful

> We might want to rename it to something else.

ya, `BatchAndWrapper` ?

**Reply by PaulRBerg** [reply]
I agree that it should be just one contract, but we can find a name later.

**Comment by PaulRBerg** [comment]
Closing as this now can be achieved via the abstract `batch` function.

https://github.com/sablier-labs/v2-core/blob/b88ead3d7dd483f7107d56320ca01e4122e00d92/src/abstracts/Batch.sol
